### PR TITLE
Revised user model and added info to JWT:

### PR DESCRIPTION
**Pull request by jovanlekic**
- changed UserDetailsImpl ( added missing fields from User model)
 - changed SignupRequest ( added missing fields from User and changed user-role relationship)
 - changed User ( added missing fields and constraints)
 - added generating JWT from UserDetailsImpl
 - changed migration scripts to correspond with database design

### CODE REVIEW

File: travel/src/main/java/com/tx/travel/controller/AuthController.java [hunk 1]
```diff
@@ -95,13 +102,18 @@ public ResponseEntity<?> registerUser(@Valid @RequestBody final SignupRequest si
...
+            signUpRequest.getSurname(),
+            signUpRequest.getPassport_no(),
+            signUpRequest.getId_no(),
+            signUpRequest.getCost_center_id());
 
-    final Set<String> strRoles = signUpRequest.getRole();
+    final String strRole = signUpRequest.getRole();
 
-    final Set<Role> roles = authService.addRoles(strRoles);
+    final Role role = authService.addRole(strRole); // mapiram String -> Role
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/controller/AuthController.java]: remove comment pls

File: travel/src/main/java/com/tx/travel/model/User.java [hunk 2]
```diff
@@ -1,59 +1,77 @@
...
-  @JoinTable(
-      name = "user_roles",
-      joinColumns = @JoinColumn(name = "user_id"),
-      inverseJoinColumns = @JoinColumn(name = "role_id"))
-  private Set<Role> roles = new HashSet<>();
+  @NotBlank
+  @Size(min = 8, max = 9)
+  private String id_no;
+
+  @NotNull private Long cost_center_id;
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/model/User.java]: use camel case for variable names everywhere

File: travel/src/main/java/com/tx/travel/model/User.java [hunk 3]
```diff
@@ -88,11 +106,51 @@ public void setPassword(String password) {
...
+
+  public String getSurname() {
+    return surname;
+  }
+
+  public void setSurname(String surname) {
+    this.surname = surname;
+  }
+
+  public String getPassport_no() {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/model/User.java]: camel case im method names please!

File: travel/src/main/java/com/tx/travel/model/User.java [hunk 4]
```diff
@@ -88,11 +106,51 @@ public void setPassword(String password) {
...
+    return passport_no;
+  }
+
+  public void setPassport_no(String passport_no) {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/model/User.java]: camel case please!

File: travel/src/main/java/com/tx/travel/model/User.java [hunk 5]
```diff
@@ -88,11 +106,51 @@ public void setPassword(String password) {
...
+    this.passport_no = passport_no;
+  }
+
+  public String getId_no() {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/model/User.java]: camel case im method names please!

File: travel/src/main/java/com/tx/travel/model/User.java [hunk 6]
```diff
@@ -88,11 +106,51 @@ public void setPassword(String password) {
...
+    return id_no;
+  }
+
+  public void setId_no(String id_no) {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/model/User.java]: camel case im method names please!

File: travel/src/main/java/com/tx/travel/model/User.java [hunk 7]
```diff
@@ -88,11 +106,51 @@ public void setPassword(String password) {
...
+    this.id_no = id_no;
+  }
+
+  public Long getCost_center_id() {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/model/User.java]: camel case im method names please!

File: travel/src/main/java/com/tx/travel/model/User.java [hunk 8]
```diff
@@ -88,11 +106,51 @@ public void setPassword(String password) {
...
+    return cost_center_id;
   }
 
-  public void setRoles(Set<Role> roles) {
-    this.roles = roles;
+  public void setCost_center_id(Long cost_center_id) {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/model/User.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java [hunk 9]
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
...
-  private Set<String> role;
+  private String role;
 
   @NotBlank
   @Size(min = 6, max = 40)
   private String password;
 
+  @NotBlank
+  @Size(min = 8, max = 10)
+  private String passport_no;
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java [hunk 10]
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
...
+
+  @NotBlank
+  @Size(min = 8, max = 9)
+  private String id_no;
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java [hunk 11]
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
...
+
+  @NotBlank
+  @Size(max = 20)
+  private String name;
+
+  @NotBlank
+  @Size(max = 20)
+  private String surname;
+
+  @NotNull private Long cost_center_id;
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java [hunk 12]
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
...
+
+  public String getPassport_no() {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java [hunk 13]
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
...
+    return passport_no;
+  }
+
+  public void setPassport_no(String passport_no) {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java [hunk 14]
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
...
+    this.passport_no = passport_no;
+  }
+
+  public String getId_no() {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java [hunk 15]
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
...
+    return id_no;
+  }
+
+  public void setId_no(String id_no) {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java [hunk 16]
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
...
+
+  public String getSurname() {
+    return surname;
+  }
+
+  public void setSurname(String surname) {
+    this.surname = surname;
+  }
+
+  public Long getCost_center_id() {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java [hunk 17]
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
...
+    return cost_center_id;
+  }
+
+  public void setCost_center_id(Long cost_center_id) {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/response/UserInfoResponse.java [hunk 18]
```diff
@@ -6,15 +6,76 @@ public class UserInfoResponse {
   private Long id;
   private String username;
   private String email;
+
+  private String passport_no;
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/response/UserInfoResponse.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/response/UserInfoResponse.java [hunk 19]
```diff
@@ -6,15 +6,76 @@ public class UserInfoResponse {
...
+  private String id_no;
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/response/UserInfoResponse.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/response/UserInfoResponse.java [hunk 20]
```diff
@@ -6,15 +6,76 @@ public class UserInfoResponse {
...
+  private String name;
+  private String surname;
+  private Long cost_center_id;
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/response/UserInfoResponse.java]: camel case in method and variable names please!

File: travel/src/main/java/com/tx/travel/payload/response/UserInfoResponse.java [hunk 21]
```diff
@@ -6,15 +6,76 @@ public class UserInfoResponse {
...
     this.email = email;
+    this.passport_no = passport_no;
+    this.id_no = id_no;
+    this.name = name;
+    this.surname = surname;
+    this.cost_center_id = cost_center_id;
     this.roles = roles;
   }
 
+  public String getPassport_no() {
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/payload/response/UserInfoResponse.java]: camel case in method and variable names please! - apply to all methods and variables in all classes

File: travel/src/main/java/com/tx/travel/security/jwt/JwtUtils.java [hunk 22]
```diff
@@ -40,7 +42,15 @@ public String getJwtFromCookies(HttpServletRequest request) {
   }
 
   public ResponseCookie generateJwtCookie(UserDetailsImpl userPrincipal) {
-    String jwt = generateTokenFromUsername(userPrincipal.getUsername());
+    String jwt = generateTokenFromUser(userPrincipal);
+    return ResponseCookie.from(jwtCookie, jwt)
+        .path("/api")
+        .maxAge(24 * 60 * 60L)
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/security/jwt/JwtUtils.java]: can we make this configurable?

File: travel/src/main/java/com/tx/travel/security/jwt/JwtUtils.java [hunk 23]
```diff
@@ -75,9 +86,18 @@ public boolean validateJwtToken(String authToken) {
     return false;
   }
 
-  public String generateTokenFromUsername(String username) {
+  public String generateTokenFromUser(UserDetailsImpl userPrincipal) {
+    Map<String, Object> claims = new HashMap<>();
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/security/jwt/JwtUtils.java:97]: can they be String?

File: travel/src/main/java/com/tx/travel/security/services/UserDetailsImpl.java [hunk 24]
```diff
@@ -19,6 +19,12 @@ public class UserDetailsImpl implements UserDetails {
 
   private String email;
 
+  private String passport_no;
```
Comment (megicivovic) [inline travel/src/main/java/com/tx/travel/security/services/UserDetailsImpl.java]: camel case in method and variable names please!

**Review by megicivovic — CHANGES_REQUESTED** [review]

File: travel/src/main/java/com/tx/travel/security/jwt/JwtUtils.java (hunk 23 above)
Comment (jovanlekic) [reply travel/src/main/java/com/tx/travel/security/jwt/JwtUtils.java:97]: .setClaims() from Jwts package wouldn't be able to resolve Map<String,String> , accepts only Map<String, Object>

File: travel/src/main/java/com/tx/travel/security/jwt/JwtUtils.java (hunk 22 above)
Comment (jovanlekic) [reply travel/src/main/java/com/tx/travel/security/jwt/JwtUtils.java]: Can you be more precise? Do you mean adding more properties to response cookie, as '.secure()' for example or changing path to be less general "./api/..."

Comment (megicivovic) [reply travel/src/main/java/com/tx/travel/security/jwt/JwtUtils.java]: the comment is referring to line .maxAge,so I mean this number should be part of application, properties and injected here as a value. But now that you mentioned it, "/api" should be a constant at least, if not also part of configuration. Always avoid using hard coded values!

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 25]
```diff
@@ -0,0 +1,132 @@
...
+
+        assertEquals(result, rola);
+    }
+
+    @Test
+    public void addRole_retOM_success(){
+
+        final String role = "mod";
+
+        final Role rola = new Role(ERole.ROLE_OFFICE_MANAGER);
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]: use english names

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 26]
```diff
@@ -0,0 +1,132 @@
...
+        final String usr = "user1";
+
+        when(userRepository.findByUsername(usr))
+                .thenReturn(Optional.of(new User(usr, "email", "123123123", "User 1", "Userovic", "123123123", "123123123", 1l)));
+
+        assertThrows(UsernameAlreadyExistsException.class,()->{ sut.findByUsernameOrEmail(usr,"email"); });
+    }
+
+    @Test
+    public void findByUsernameOrEmail_thrownEmailAlreadyExistedException_success(){
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]:
```suggestion
    public void findByUsernameOrEmail_thrownEmailAlreadyExistedException(){
```

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 27]
```diff
@@ -0,0 +1,132 @@
...
+    @Mock
+    UserRepository userRepository;
+
+    @Mock
+    RoleRepository roleRepository;
+
+    @InjectMocks
+    AuthService sut;
+
+    @DisplayName("given auth service" +
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]: rephrase please, given is condition, then is expected result

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 28]
```diff
@@ -0,0 +1,132 @@
...
+        final String email = "usr1@email.com";
+
+        when(userRepository.findByEmail(email))
+                .thenReturn(Optional.of(new User("usr", email, "123123123", "User 1", "Userovic", "123123123", "123123123", 1l)));
+
+        assertThrows(EmailAlreadyExistsException.class,()->{ sut.findByUsernameOrEmail("",email); });
+    }
+
+    @Test
+    public void addRoleAdmin_retAdmin_success(){
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]:
```suggestion
    public void addRoleAdmin_success(){
```

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 29]
```diff
@@ -0,0 +1,132 @@
...
+        when(roleRepository.findByName(ERole.ROLE_ADMIN))
+                .thenReturn(Optional.of(rola));
+
+        final Role result = sut.addRoleAdmin(role);
+
+        assertEquals(result, rola);
+    }
+
+    @Test
+    public void addRole_retOM_success(){
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]:
```suggestion
    public void addRoleOM_success(){
```

Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]: add display name

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 30]
```diff
@@ -0,0 +1,132 @@
...
+        when(roleRepository.findByName(ERole.ROLE_OFFICE_MANAGER))
+                .thenReturn(Optional.of(rola));
+
+        final Role result = sut.addRoleAdmin(role);
+
+        assertEquals(result, rola);
+    }
+
+    @Test
+    public void addRole_retEmpl_success(){
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]: add display name

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 31]
```diff
@@ -0,0 +1,132 @@
...
+
+        final String role = "afa"; //nesto sto nije admin/mod
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]:
```suggestion
        final String role = "test_role";
```

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java (hunk 28 above)
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]: add display name

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 32]
```diff
@@ -0,0 +1,132 @@
...
+
+        final Role rola = new Role(ERole.ROLE_EMPLOYEE);
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]: english names

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 33]
```diff
@@ -0,0 +1,132 @@
...
+        when(roleRepository.findByName(ERole.ROLE_EMPLOYEE))
+                .thenReturn(Optional.of(rola));
+
+        final Role result = sut.addRoleAdmin(role);
+
+        assertEquals(result, rola);
+    }
+
+    @Test
+    public void addRole_retEmplNull_success(){
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]:
```suggestion
    public void addRole_null{
```

Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]: add display name for this test

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 34]
```diff
@@ -0,0 +1,132 @@
...
+
+        final String role = null;//nesto sto nije admin/mod
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]:
```suggestion
        final String role = null;
```

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 35]
```diff
@@ -0,0 +1,132 @@
...
+
+        final Role rola = new Role(ERole.ROLE_EMPLOYEE);
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]: english names

**Review by megicivovic — CHANGES_REQUESTED** [review]

File: travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java [hunk 36]
```diff
@@ -15,12 +16,72 @@ public class SignupRequest {
...
 
-  private Set<String> role;
-
   @NotBlank
   @Size(min = 6, max = 40)
   private String password;
 
+  @NotBlank
+  @Size(min = 8, max = 10)
+  @Column(name="passport_no")
```
Comment (petkocfc) [inline travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java]: Why do we have @Column in Request?

File: travel/src/main/java/com/tx/travel/service/AuthService.java [hunk 37]
```diff
@@ -25,55 +24,47 @@ public AuthService(final UserRepository userRepository, final RoleRepository rol
...
     }
   }
 
-  public Set<Role> addRoles(final Set<String> strRoles) {
-    final Set<Role> roles = new HashSet<>();
-    if (strRoles == null) {
-      final Role userRole =
-          roleRepository
+  public Role addRoleAdmin(final String strRole) {
+    if (strRole == null) {
```
Comment (petkocfc) [inline travel/src/main/java/com/tx/travel/service/AuthService.java]: this method should be refactored

File: travel/src/main/java/com/tx/travel/service/AuthService.java [hunk 38]
```diff
@@ -25,55 +24,47 @@ public AuthService(final UserRepository userRepository, final RoleRepository rol
...
-                roles.add(userRole);
-            }
-          });
+        }
+      }
     }
-    return roles;
+  }
+
+  public Role addRole(final ERole role) {
```
Comment (petkocfc) [inline travel/src/main/java/com/tx/travel/service/AuthService.java]: here we have a method addRole and we are fetching the role, shouldn't it be `getRole` or?

File: travel/src/main/resources/db/migration/V4__drop_user_roles.sql [hunk 39]
```diff
@@ -0,0 +1 @@
+DROP TABLE IF EXISTS `user_roles`;
```
Comment (petkocfc) [inline travel/src/main/resources/db/migration/V4__drop_user_roles.sql]: add newline

File: travel/src/main/resources/db/migration/V5__add_columns_user.sql [hunk 40]
```diff
@@ -0,0 +1,8 @@
+ALTER TABLE `users`
+    ADD COLUMN IF NOT EXISTS `name` varchar(20) DEFAULT NULL,
+    ADD COLUMN IF NOT EXISTS `surname` varchar(20) DEFAULT NULL,
+    ADD COLUMN IF NOT EXISTS `passport_no` varchar(20) DEFAULT NULL,
+    ADD COLUMN IF NOT EXISTS `id_no` varchar(20) DEFAULT NULL,
+    ADD COLUMN IF NOT EXISTS `cost_center_id` bigint(20) NOT NULL,
+    ADD COLUMN IF NOT EXISTS `role_id` int(11) NOT NULL,
+    ADD CONSTRAINT `FK_roles_role_id_constraint` FOREIGN KEY (`role_id`) REFERENCES `roles` (`id`);
```
Comment (petkocfc) [inline travel/src/main/resources/db/migration/V5__add_columns_user.sql]: add newline

File: travel/src/main/java/com/tx/travel/service/AuthService.java [hunk 41]
```diff
@@ -25,55 +24,47 @@ public AuthService(final UserRepository userRepository, final RoleRepository rol
...
-      throws UsernameAlreadyExistsException {
+      throws UsernameAlreadyExistsException,EmailAlreadyExistsException {
     final Optional<User> userByUsername = userRepository.findByUsername(username);
     if (userByUsername.isPresent()) {
       throw new UsernameAlreadyExistsException(username);
     }
     final Optional<User> userByEmail = userRepository.findByEmail(email);
     if (userByEmail.isPresent()) {
-      throw new UsernameAlreadyExistsException(email);
+      throw new EmailAlreadyExistsException(email);
```
Comment (petkocfc) [inline travel/src/main/java/com/tx/travel/service/AuthService.java:34]: method is called `findByUsernameOrEmail` and we are returning exception `EmailAlreadyExistsException` should not work like this?

**Review by petkocfc — APPROVED** [review]

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 42]
```diff
@@ -1,149 +1,131 @@
...
+                    "123123123",
+                    "123123123",
+                    1L)));
+
+    assertThrows(
+        EmailAlreadyExistsException.class,
+        () -> sut.findByUsernameOrEmail("", email));
+  }
+
+  @DisplayName("given string admin" + " when role was searched" + " then return admin role")
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]: split in separate rows

File: travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java [hunk 43]
```diff
@@ -1,149 +1,131 @@
...
+    final Role role = new Role(ERole.ROLE_ADMIN);
+    role.setId(2);
+
+    when(roleRepository.findByName(ERole.ROLE_ADMIN)).thenReturn(Optional.of(role));
+
+    final Role result = sut.getRole(ERole.ROLE_ADMIN);
+    assertEquals(result, role);
+  }
+
+  @DisplayName("given string mod" + " when role was searched" + " then return office manager role")
```
Comment (megicivovic) [inline travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java]: split in separate rows

**Review by megicivovic — COMMENTED** [review]
Almost there, just be mindfull of the formatting!

**Review by megicivovic — APPROVED** [review]
Amazing work! 🚀

**Pull request merged**